pkg/book_service/book_service.go

```diff
@@ -0,0 +1,118 @@
+package book_Service
+
+import (
+	"encoding/json"
+	action "github.com/daniyakubov/book_service_n/pkg/action"
+	"github.com/daniyakubov/book_service_n/pkg/book_service/models"
+	"github.com/daniyakubov/book_service_n/pkg/cache"
+	"github.com/daniyakubov/book_service_n/pkg/elastic_service"
+	errors "github.com/fiverr/go_errors"
+	"strings"
+)
+
+type BookService struct {
+	booksCache     cache.Cache
+	elasticHandler *elastic_service.ElasticHandler
+}
+
+func NewBookService(booksCache cache.Cache, elasticHandler *elastic_service.ElasticHandler) BookService {
+	return BookService{
+		booksCache:     booksCache,
+		elasticHandler: elasticHandler,
+	}
+}
+
+func (b *BookService) PutBook(req *models.Request) (*models.PutResponse, error) {
```

Comment:
put is a very http term - at the end you declare the functionality of "add book"
```suggestion
func (b *BookService) AddBook(req *models.Request) (*models.PutResponse, error) {
```